pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -465,6 +465,12 @@ func (r *reconciler) currentLoadBalancerService(ci *operatorv1.IngressController
 		}
 		return false, nil, err
 	}
+
+	// BZ2054200: Don't delete services that are not directly owned by this controller
+	if service.Labels[manifests.OwningIngressControllerLabel] != ci.Name {
```

Comment:
`currrentLoadBalancer` is called by other methods that are not deleting.  I think you need to check this only when deleting.

Reply:
Correct - this method determines whether we have a `currentLoadBalancerService`. Either deleting, creating, or updating, we still want to filter out these services that aren't owned by our ingress controller, so I'd think it makes sense to keep this logic generic.

Reply:
If this api call can return multiple services, then you need to query for a list and iterate over a list.

Reply:
Hmmm - I don't think that's possible with r.client.Get since names are unique.

Reply:
Isn't there an r.client.List?

Reply:
Do you agree that it's possible to find more than once service attached to this ic?

Reply:
I believe this is what you need to use: 
https://github.com/kubernetes/client-go/blob/master/listers/core/v1/service.go#L82

Reply:
Hmm, I don't believe so. We are looking for a single service that the load_balancer_service.go with a unique name; there wouldn't ever be more than 1 service with the same name in the same namespace.

But I think I see potentially where you are headed. If the loadbalancer service already exists, but isn't owned by the ingress controller, then it will declare it doesn't have a service. The operator will go attempt to create it, but that create will fail because it already exists, leaving it in a broken state. I will fix that.

Reply:
Ah yea sorry! You were correct with your first comment. I think we only want to check when we delete otherwise this causes issues. I think we are still looking for a singular service object, I just had everything backwards.

Reply:
Okay there is definitely some nuance to this. I pushed up a new update, it wasn't quite that straight forward because if you put only wrapped the delete logic to avoid deleting, then it would indefinitely hang the deletion of the ingress controller because it still thought it had a service.

Also, if a service existed, the ingress controller would try to fix it and inevitably put it in a broken state. So it now just throws an error if you have an existing service with a name collision and it won't touch it.